Some fix and improvements

Xseuguh opened the pull request:
Use of a Mat to store current modification, so we can save one call to applyManipulation

Better error catching when use cvtColor

Add a method removeOption to clear the option menu

Change an image during a manipulation cancel it now :)

Xseuguh committed:
Use of a Mat to store current modification, so we can save one call to applyManipulation

Better error catching when use cvtColor

Add a method removeOption to clear the option menu

Change an image during a manipulation cancel it now :)

Yakiigit commented on BetterThanGIMP/src/manipulations/CannyEdge/CannyEdge.cpp:
```diff
@@ -50,7 +50,11 @@ CannyEdge::CannyEdge(Workspace &w) : Manipulation(w) {
 Mat CannyEdge::cannyEdge(Mat &image, int threshold1, int threshold2, int kernelSize) {
     Mat result, grey, blur;
-    cvtColor(image, grey, COLOR_BGR2GRAY);
+    if (image.channels() > 3) {
```
Isn't ">=3" more appropriate than ">3"?

Xseuguh committed:
Some fixes

Remove grey conversion before Canny, looks like opencv is doing it already

Initialize imageModified in manipulation to prevent errors
